rust/core-lib/src/editor.rs

```diff
@@ -762,13 +757,12 @@ impl<W: Write + Send + 'static> Editor<W> {
     }
 
     pub fn do_rpc(self_ref: &Arc<Mutex<Editor<W>>>, view_id: &str, cmd: EditCommand) -> Option<Value> {
-        self_ref.lock().unwrap().do_rpc_with_self_ref(view_id, cmd, self_ref)
+        self_ref.lock().unwrap().do_rpc_impl(view_id, cmd)
     }
 
-    fn do_rpc_with_self_ref(&mut self, view_id: &str,
-                  cmd: EditCommand,
-                  self_ref: &Arc<Mutex<Editor<W>>>)
-                  -> Option<Value> {
+    //FIXME: what is the relationship between this and `do_rpc`, now?
```

Comment:
we've moved the arc/mutex up a level; instead of wrapping each editor, it wraps all editors. Editors no longer call plugins directly, so they don't need self_ref.